test/test_nn.py

```diff
@@ -888,6 +888,81 @@ def test_invalid_conv3d(self):
+                          stride=2, dilation=2)
+        self.assertEqual(expect, module(x))
+
+        # non-zero padding modes require explicit padding
+        module = nn.Conv1d(in_channels=1, out_channels=1, kernel_size=10,
+                           padding='same', padding_mode='reflect')
+        x_padded = F.pad(x, [4, 5], mode='reflect')
+        expect = F.conv1d(x_padded, module.weight, module.bias, padding='valid')
+        self.assertEqual(expect, module(x))
+        self.assertEqual(x.size(), expect.size())
+
+        # Invalid padding string
+        with self.assertRaisesRegex(ValueError, 'Invalid padding string'):
+            module = nn.Conv1d(in_channels=3, out_channels=33, kernel_size=10, padding='foo')
+
+    def test_Conv2d_module_same_padding(self):
+        x = torch.rand(1, 1, 9, 20)
+        module = nn.Conv2d(in_channels=1, out_channels=1, kernel_size=(5, 10),
+                           padding='same')
+        expect = F.conv2d(x, module.weight, module.bias, padding='same')
+        self.assertEqual(expect, module(x))
+
+        module = nn.Conv2d(in_channels=1, out_channels=1, kernel_size=(3, 4),
+                           padding='same', stride=(2, 3), dilation=(1, 2))
+        expect = F.conv2d(x, module.weight, module.bias, padding='same',
+                          stride=(2, 3), dilation=(1, 2))
+        self.assertEqual(expect, module(x))
+
+        # non-zero padding modes require explicit padding
+        module = nn.Conv2d(in_channels=1, out_channels=1, kernel_size=(3, 4),
+                           padding='same', padding_mode='reflect')
+        x_padded = F.pad(x, [1, 2, 1, 1], mode='reflect')
+        expect = F.conv2d(x_padded, module.weight, module.bias, padding='valid')
+        self.assertEqual(expect, module(x))
+        self.assertEqual(x.size(), expect.size())
+
+        # Invalid padding string
+        with self.assertRaisesRegex(ValueError, 'Invalid padding string'):
+            module = nn.Conv2d(in_channels=3, out_channels=33, kernel_size=10, padding='foo')
+
+    def test_Conv3d_module_same_padding(self):
+        x = torch.rand(1, 1, 4, 4, 4)
+        module = nn.Conv3d(in_channels=1, out_channels=1, kernel_size=(2, 3, 4),
+                           padding='same')
+        expect = F.conv3d(x, module.weight, module.bias, padding='same')
+        self.assertEqual(expect, module(x))
+
+        module = nn.Conv3d(in_channels=1, out_channels=1, kernel_size=(2, 3, 4),
+                           padding='same', stride=(1, 2, 3), dilation=(3, 2, 1))
+        expect = F.conv3d(x, module.weight, module.bias, padding='same',
+                          stride=(1, 2, 3), dilation=(3, 2, 1))
+        self.assertEqual(expect, module(x))
+
+        # non-zero padding modes require explicit padding
+        module = nn.Conv3d(in_channels=1, out_channels=1, kernel_size=(2, 3, 4),
+                           padding='same', padding_mode='circular')
```

Comment:
nit: I see tests for `circular`, `reflect`, and (implicitly) `zeros` padding modes, but none for `replicate`. Please add at least one test that covers the `replicate` padding mode with `same` padding.